code/game/objects/items/misc.dm

```diff
@@ -50,25 +38,124 @@
 	item_state = "gift"
 	w_class = SIZE_LARGE
 
-/obj/item/staff/gentcane
-	name = "Gentlemans Cane"
-	desc = "An ebony cane with an ivory tip."
+/obj/item/weapon/melee/pole
+	name = "wooden pole"
+	desc = "A rough, cracked pole seemingly constructed on the field. You could probably whack someone with this."
 	icon = 'icons/obj/items/weapons/weapons.dmi'
-	icon_state = "cane"
-	item_state = "stick"
+	icon_state = "wooden_pole"
+	item_state = "wooden_pole"
+	force = 20
+	attack_speed = 1.5 SECONDS
+	var/gripped = FALSE
 
-/obj/item/staff/stick
-	name = "stick"
-	desc = "A great tool to drag someone else's drinks across the bar."
-	icon = 'icons/obj/items/weapons/weapons.dmi'
-	icon_state = "stick"
-	item_state = "stick"
-	force = 3
-	throwforce = 5
-	throw_speed = SPEED_FAST
-	throw_range = 5
-	w_class = SIZE_SMALL
-	flags_item = NOSHIELD
+/obj/item/weapon/melee/pole/get_examine_text(mob/user)
+	. = ..()
+	. += SPAN_NOTICE("Activate on your hand to grip this tightly. Useful if you have a bad leg.")
+
+/obj/item/weapon/melee/pole/attack_self(mob/living/user)
+	..()
+	if(!gripped)
+		user.visible_message(SPAN_NOTICE("[user] grips [src] tightly."), SPAN_NOTICE("You grip [src] tightly."))
+		flags_item |= NODROP
+		ADD_TRAIT(user, TRAIT_HOLDS_CANE, TRAIT_SOURCE_ITEM)
+		user.AddComponent(/datum/component/footstep, 6, 25, 4, 1, "cane_step")
+		gripped = TRUE
+	else
+		user.visible_message(SPAN_NOTICE("[user] loosens \his grip on [src]."), SPAN_NOTICE("You loosen your grip on [src]."))
+		flags_item &= ~NODROP
+		REMOVE_TRAIT(user, TRAIT_HOLDS_CANE, TRAIT_SOURCE_ITEM)
+		// Ideally, this would be something like a component added onto every mob that prioritizes certain sounds, such as stomping over canes.
+		var/component = user.GetComponent(/datum/component/footstep)
```

Comment:
to de-shitcode (slightly) this, you could make a root mob proc called stop_footsteps that sends a signal like COMSIG_MOB_STOP_FOOTSTEPS which /datum/component/footstep starts listening to when it's initialized and causes it to clean itself up

Reply:
there is no reason to do that, it just wraps the shitcode in a confusing blanket that future devs will consider good code/ not realize sucks

Reply:
components detaching themselves on signals is literally how ur meant to do it.............................. but fine